### ISSUE TITLE
Associations doesn't work

### ISSUE THREAD

**Issue by juanda645** [jspizziri/sequelize-connect#16]
I create a model like this:
```js
// user.model.js
"use strict";

module.exports = function(sequelize, DataTypes) {
  var User = sequelize.define("User", {
    username: DataTypes.STRING
  }, {
    classMethods: {
      associate: function(models) {
        User.hasMany(models.Task)
      }
    }
  });

  return User;
};
```
But the association doesn't appear in the data base, what i doing wrong?

**Comment by jspizziri** [comment]
@juanda645,

Sorry for the late response, for some reason I didn't see the notification. Are you sure that `user.model.js` is in a discoverable path when you configured `sequelize-connect`?

e.g. If I do the following:

```js
// app.js
var Connection 		= require('sequelize-connect');

var discover = [__dirname + '/models', ...];
var orm = new Connection(
  'test-db',
  'test-user',
  'secret1234',
  {
    dialect: "mysql",
    port:    3306
  },
  discover,
)
.then(function(instance){
  // Connection is completed
});
```

then `user.model.js` needs to be within `var discover = [__dirname + '/models', ...];`.

**Comment by juanda645** [comment]
Yes, it is in the correct path because the user table is created in the database, if wasn't it the user table wouldn't exist.

**Comment by jspizziri** [comment]
@juanda645 ,

and is the `associate` method being called? Can you do something like the following:

```js
associate: function(models) {
    // debug log line
    console.debug('associating models: ', User, models.Task);
    User.hasMany(models.Task)
}
```

**Comment by juanda645** [comment]
I tried that but the associate method was never called. But by team decision I had to change to knex, so I don't use sequelize anymore, but would be good if you check that issue.

**Comment by jspizziri** [comment]
@juanda645 ,

Thanks for the update. If you need any further help please let me know. If anyone else has this same issue please let me know and I'll reopen it if we can get more information on what's going on.

**Comment by jspizziri** [comment]
Actually it looks like it's an issue with the version of sequelize: https://github.com/sequelize/sequelize/issues/7804#issuecomment-309665343.

I'll try to fix this in the next couple of days and lock down the sequelize version in the package file.

**Comment by absolute-fx** [comment]
Exact same thing here. hasMany doesn't work. However, BelongsToMany is working well.
Any chance to get this fixed soon?

**Comment by jspizziri** [comment]
@absolute-fx , sorry been a bit busy and this fell off my radar. I'll try to get to it this weekend. However, if you're open to it I'd gladly accept a PR!

**Comment by absolute-fx** [comment]
@jspizziri , ok I've spent hours on this but I've finaly found the solution :)
As referenced here http://docs.sequelizejs.com/manual/tutorial/upgrade-to-v4.html, the classMethods doesn't exist anymore in V4.
Just adding this in the model works nice and smooth:

> Model.associate = function (models) {...}

you can write the connection like this (without the promise):

> const Connection = require('sequelize-connect');
const discover = [__dirname + '/models'];
const dbConfig = require(__dirname + '/../db_login.json');
module.exports =  new Connection(
    dbConfig.db,
    dbConfig.login,
    dbConfig.pass,
    {
        host: dbConfig.host,
        dialect: "mysql",
        port:    3306,
        dialectOptions: {
            charset: 'utf8',
        }
    },
    discover
);

and then in your model:

> module.exports = function (sequelize, DataTypes) {
var Models = sequelize.define('Models', {....});
Models.associate = function (models) {...}
return Models;
};

Et voilà ;)
Just update your README.

**Comment by jspizziri** [comment]
Just published v2.1.0, which is intended to resolve this issue: https://github.com/jspizziri/sequelize-connect/commit/6c9d2914caad0cdd49e423a758ff398df84889e3
